src/extension/shell.rs

```diff
@@ -171,6 +171,19 @@ impl super::Extension for ShellExtension {
                     warn!("Shell command reported error: \n{}", error_str);
                 }
 
+                // Check if debug flag set, provide additional context when an error occurs.
+                let debug_opt = params.get(&Value::from("debug"));
+                let with_debug = if let Some(value) = debug_opt {
+                    let val = value.as_bool();
+                    val.unwrap_or(false)
+                }else{
+                    false
+                };
+                
+                if with_debug {
+                    error!("debug for shell cmd '{}', exit_status '{}', stdout '{}', stderr '{}'", cmd, output.status, output_str, error_str);
```

Comment:
Could you please use info!() instead of `error`?

Reply:
Yes no problem, and thanks for clarifying on merging into `dev` first.